README.md

```diff
@@ -20,9 +19,38 @@ A cloud-based agent that forwards messages to mobile devices.
+
+- Establishing logical connections: The Mediator enables entities (such as individuals or organizations) to establish secure connections with each other. Each entity has a unique DID that serves as its identifier on the decentralized network.
+- DID resolution: When an entity wants to communicate with another entity, it resolves the recipient's DID to obtain the necessary information to establish a connection. The resolution process involves retrieving the recipient's public key and associated metadata from a decentralized identity infrastructure, such as a blockchain or a distributed ledger.
+- Message encryption: The sender encrypts the message two times (one for the final receiver and then warp the encrypted message and encrypt again to Mediator) using the mediator's and recipient's public key obtained during the DID resolution process. This ensures that only the intended recipient can decrypt and read the message.
+
+```mermaid
+graph LR
+  subgraph Encryted message to Mediator
+    subgraph Encryted message to Reciever
+    id1[[The planetext message]]
+    end
+  end
+```
+
+- Message routing: The sender sends the encrypted message to the Mediator. The Mediator acts as a routing agent, receiving messages from one entity, decrypted one layer and forwarding them to the intended recipient based on their DID.
+- Mediation process: The Mediator verifies the authenticity and integrity of the incoming message by checking the digital signature attached to it. This signature ensures that the message was indeed sent by the claimed sender and that it hasn't been tampered with during transmission.
+
+- Message decryption: After verifying the message's authenticity, the Mediator decrypted one layer of the message using the mediator's private key, which is securely held by the mediator. Once decrypted, the next message becomes readable (the final planetext intended for the final user it's still encrypted).
+- Optional processing: The Mediator may perform additional processing on the message based on predefined rules or business logic. This could include applying filters, applying policies, or invoking external services.
+- Message forwarding: If necessary, the Mediator can further forward the decrypted message to additional entities in the communication flow. This enables multi-party communication scenarios.
+
+By acting as an intermediary, the DID Comm v2 Mediator helps facilitate secure and private communication between entities while leveraging the decentralized nature of DIDs and cryptographic techniques to ensure authenticity, integrity, and confidentiality of the messages exchanged.
+
+The mediator is especially useful when the edge entities are not always online, like mobile devices. Usually, we can assume that the mediator is always online.
+
```

Comment:
Below i got revised from chat GPT
--------------------------------------- 
Message routing: The sender transmits an encrypted message to the Mediator, which serves as a routing agent. In this role, the Mediator receives messages from the sender, decrypts one layer, and forwards them to the appropriate recipient based on the recipient’s DID.

Mediation process: The Mediator validates the authenticity and integrity of the incoming message by examining the attached digital signature. This verification process ensures that the message was genuinely sent by the sender it purports to come from and that it has not been altered during transmission.

Message decryption: Upon confirming the authenticity of the message, the Mediator decrypts one layer of encryption using its private key, which is securely stored. This decryption makes the next layer of the message readable, although the final plaintext intended for the end user remains encrypted.

Optional processing: The Mediator has the capability to conduct additional processing on the message according to predefined criteria or business logic. This may include applying filters, enforcing policies, or utilizing external services.

Message forwarding: When required, the Mediator can also relay the decrypted message to further participants in the communication process. This is particularly useful in scenarios involving communication among multiple parties.

As an intermediary, the DID Comm v2 Mediator is instrumental in ensuring secure and private communication between entities. By capitalizing on the decentralized nature of DIDs coupled with cryptographic techniques, it guarantees the authenticity, integrity, and confidentiality of the messages being exchanged.

Additionally, the Mediator proves to be invaluable in instances where edge entities may not maintain a consistent online presence, such as mobile devices. In contrast, the Mediator is typically expected to be online at all times.